### DISCUSSION TITLE
Schedule once an hour, but do something different once a day?

### DISCUSSION THREAD

**Discussion by simonw** [community/community#26814] [Actions]
<p>I have a task that runs hourly - but I want to run it very slightly differently once a day: I want to skip one of the steps in it every 24 hours.</p>
<p>Any neat mechanisms for me to do that?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/simonw">@simonw</a> ,</p>
<p>You can try the following configurations in your workflow:</p>
<pre><code>on:
  schedule:
    - cron: '0 1-23 * * *'
    - cron: '0 0 * * *'

jobs:
  test_schedule:
    name: Test schedule
    runs-on: ubuntu-latest
    steps:
      - name: Skip this step every 24 hours
        if: github.event_name == 'schedule' &amp;&amp; github.event.schedule != '0 0 * * *'
        run: echo "This step will be skipped every 24 hours"
</code></pre>
<p><strong>1)</strong> The first cron ’ <strong>0 1-23 * * *</strong>’ will trigger the workflow hourly from the 1st hour to 23th hour.</p>
<p><strong>2)</strong> The second cron ’ <strong>0 0 * * *</strong>’ will only trigger the workflow at 00:00 UTC.</p>
<p><strong>3)</strong> The property " <strong>github.event.schedule</strong>" of the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a> returns the cron of the schedule that triggers the current workflow run.</p>
<p><strong>4)</strong> On the step that need to be skipped every 24 hours, set the <strong>if conditional</strong>  to check which cron of the schedule triggers the current workflow run.  If the cron is not ’ <strong>0 0 * * *</strong>’, execute this step, otherwise skip this step.</p>

**Comment by simonw** [comment]
<p>Thanks very much - this looks like exactly the solution I need.</p>

**Comment by simonw** [comment]
<p>It worked! <a href="https://github.com/simonw/big-local-datasette/issues/11" rel="nofollow noopener">https://github.com/simonw/big-local-datasette/issues/11</a></p>

**Comment by jrfnl** [comment]
<p>Just wanted to say thank <a class="mention" href="/brightran">@brightran</a>! Was looking to figure out the payloads available on <code>github.event</code> and this helped me no end!</p>
